### PR TITLE
Emmet in typescript styled plugin

### LINKED ISSUE #34: Add emmet support
From #33

Feature request to add support for emmet inside of styled components strings.

### PR DESCRIPTION
**Pull request by ramya-rao-a**
- Use `vscode-emmet-helper` to get emmet completions at the right places
- Add tests based on current completion tests
- Add emmet configurations 

Fixes #34

### CODE REVIEW

File: src/styled-template-language-service.ts [hunk 1]
```diff
@@ -240,13 +241,22 @@ export default class StyledTemplateLanguageService implements TemplateLanguageSe
...
         const completionsCss = this.cssLanguageService.doComplete(doc, virtualPosition, stylesheet) || emptyCompletionList;
         const completionsScss = this.scssLanguageService.doComplete(doc, virtualPosition, stylesheet) || emptyCompletionList;
         completionsScss.items = filterScssCompletionItems(completionsScss.items);
         const completions: vscode.CompletionList = {
             isIncomplete: false,
             items: [...completionsCss.items, ...completionsScss.items],
         };
+        if (emmetResults.items.length) {
+            completions.items.push(...emmetResults.items);
+            completions.isIncomplete = true;
```
Comment (mjbvz) [inline src/styled-template-language-service.ts]: Just a heads up: `isIncomplete` is ignored by TS so setting it to true won't have any effect. Does emmet depend on this?

File: e2e/tests/emmetCompletions.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+
+    it('should return emmet property completions for multiline string', () => {
+        const server = createServer();
+        openMockFile(server, mockFileName, [
+            'const q = css`',
+            'm10-20',
+            '`'
+        ].join('\n'));
+        server.send({ command: 'completions', arguments: { file: mockFileName, offset: 22, line: 1 } });
```
Comment (mjbvz) [inline e2e/tests/emmetCompletions.js]: [nit] Use the actual `line` numbers for the tests

File: src/styled-template-language-service.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/styled-template-language-service.ts]: yes, it does. For example: `m` -> `margin: ;` and `mt` -> `margin-top: ;`. So the completions need to be re-calculated for almost every char

Comment (mjbvz) [reply src/styled-template-language-service.ts]: Opened https://github.com/Microsoft/TypeScript/issues/21999. I've discussed this problem with TS before but emmet may be the first real use case for it.

As we discussed offline, I believe we should be able to take in this PR even without `isIncomplete`. Users will just have to manually trigger completions in some cases. Do you agree @ramya-rao-a?

Comment (ramya-rao-a) [reply src/styled-template-language-service.ts]: Agreed

**Comment by mjbvz** [comment]
Awesome! Thank you for the great work

I'll publish a 0.6 of this plugin shortly and submit a PR back to the vscode-styled-componnets extension to pick up the new version of the plugin

**Pull request merged**